lib/linguist/blob_helper.rb

```diff
@@ -289,6 +287,44 @@ def lines
         end
     end
 
+    def encoded_newlines_re
+      @encoded_newlines_re ||= Regexp.union(["\r\n", "\r", "\n"].
```

Comment:
Does the [`\R` extension](https://github.com/kkos/oniguruma/blob/master/doc/RE#L73) not work here?

I also take it Ruby's regex engine doesn't have the equivalent of Perl's `/a` modifier?

Reply:
I'm changing as little code as I can; this is just a refactor from:

https://github.com/github/linguist/blob/0b9c05f989a66e3e92ca9a1e0b236781ef54229f/lib/linguist/blob_helper.rb#L278-L279

`\R` also catches `[\v\f]` which we definitely don't want.

> I also take it Ruby's regex engine doesn't have the equivalent of Perl's `/a` modifier?

```shell
~$ ruby -e '//a'
-e:1: unknown regexp option - a
```

It doesn't, and more to the point, it wouldn't help for our use here, which isn't about Unicode-aware matching so much as avoiding terrible encoding exceptions rising from the deep. `/a` modifies the meaning of several sequences in the regular expressions itself, rather than changing how a regular expression is applied to a given byte-sequence-tagged-with-an-encoding (i.e. a `String`), whatever the meaning of its contents.

Reply:
Ah, I see. ;) Just thought to ask, since it's used very little in Perl (for good reasons). Thanks!